### ISSUE TITLE
Scylla doesn't set default validation class properly for compact tables

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#2605]
Scylla version: 1.7

CQL:
```
create table test (k int, ck int, ck1 int, v int, primary key (k, ck, ck1)) with compact storage
```

Scylla sets `default_validation_class` in schema tables to `org.apache.cassandra.db.marshal.BytesType` instead of `org.apache.cassandra.db.marshal.Int32Type`.

We don't support mixed thrift tables, so it has no impact for Scylla, but it may cause compatibility problems.

**Comment by duarten** [comment]
@tgrabiec I can't find any evidence of the `default_validation_class` being `org.apache.cassandra.db.marshal.Int32Type`; I see it as being `org.apache.cassandra.db.marshal.BytesType` going back to 2.2.9.

I also don't understand the relationship with compact tables - I thought the validator was meant for dynamic CFs where the column type is unknown ahead of time.

**Comment by duarten** [comment]
Fixed by 013fa3da14fd2ed1e60c8f3f4a793349f9728db6.
